src/DIRAC/Core/Utilities/JEncode.py

```diff
@@ -105,7 +122,9 @@ def default(self, obj):  # pylint: disable=method-hidden
         # if the object inherits from JSJerializable, try to serialize it
         elif isinstance(obj, JSerializable):
             return obj._toJSON()  # pylint: disable=protected-access
-
+        # if the object a bytes, decode it
```

Comment:
In fact, I tested some complex bytes and they are transmitted correctly here for both py 2/3.
```
data = b"x\x9c\xcbH\xcd\xc9\xc9W(\xcf/\xcaIQPT\x04\x00'\xe5\x04\xbf"
print 'src:', data
encodedData = encode(data)
print 'encoded:', encodedData
decodedData = decode(encodedData)[0]
print 'decoded:', decodedData
print decodedData == data
```
```
src: x��H���W(�/�IQPT '��
encoded: {"obj": "eJzLSM3JyVcozy/KSVFQVAQAJ+UEvw==", "__dCls": "b64"}
decoded: x��H���W(�/�IQPT '��
True
```

Here some more examples:
```
for d in ['\xc2\x80',
          '\x80',
          u'hello',
          u'\xc2\x80',
          b64encode('\xc2\x80'),
          'hello',
          b"x\x9c\xcbH\xcd\xc9\xc9W(\xcf/\xcaIQPT\x04\x00'\xe5\x04\xbf"]:

    print d == decode(encode(d))[0]
```

```
False
True
True
True
True
True
True

jdoodle.py:100: UnicodeWarning: Unicode equal comparison failed to convert both arguments to Unicode - interpreting them as being unequal
  print d == decode(encode(d))[0]
```

As you see `\xc2\x80` still changes incorrectly, but this is not due to my changes, but to the standard `json` encoder, so I left it because the improvement may violate backward compatibility.

Reply:
The complex cases will pass but the trivial ones will fail when mixing Python versions. For example sending `b'hello'` from Python 2 will be incorrectly parsed in Python 3 as `'hello'` (rather than `b'hello'`).

Reply:
```
assert encode(b"hello") == '{"__dCls": "b64", "obj": "aGVsbG8K"}'
```

Reply:
i.e. this test is what would need to pass with Python 2:
```
assert encode("hello") == 'hello'
assert encode(b"hello") == '{"__dCls": "b64", "obj": "aGVsbG8K"}'
```

Reply:
`b"hello"` as I understand it, it won't come here with python 2, it will be converted by json earlier.

Reply:
that is
```
assert encode("hello") == 'hello'
assert encode(b"hello") == 'hello'
```
I do not see now a real opportunity to sort str and bytes in python 2, that's the problem. And it seems to me that when switching to python 3, it is easier to consider this point on the side of python 3 code:
```
if not isinstance(incomingData, bytes):
   bytes(incomingData, 'utf-8')
```

Reply:
Given that:
* JEncode isn't being actively used by anyone (I think?)
* We have production quality releases for Python 3 
I think it's better to **only** have support for decoding `"__dCls" == "b64"` in Python 2. Sending binary data is just too ambiguous and not worth the effort.

In fact could we make DIRAC crash if `DIRAC_USE_JSON_ENCODE` is set while running with Python 2?

Reply:
>I think it's better to only have support for decoding "__dCls" == "b64" in Python 2. Sending binary data is just too ambiguous and not worth the effort.

I thought about it. If we say we don't need backward compatibility for `JEncode`, then it looks like an exit, but we will always get bytes, right? If so its not very good..
```
python2         |  python 3
encode(b'hello') -> decode() -> b'hello'
encode('hello')  -> decode() -> b'hello'
```

Reply:
>In fact could we make DIRAC crash if DIRAC_USE_JSON_ENCODE is set while running with Python 2?

in my opinion this whole story with `DIRAC_USE_JSON_ENCODE` just doesn't make sense in the context of string types processing and python 2. Significant changes are needed for me, including `DEncode` to align it with `JEncode`, and changing `DEncode` we are guaranteed to break everything)